injector.c

```diff
@@ -11,6 +11,8 @@
 
 #include <xc.h>
 
+static bool injector_open_called_last = false;
```

Comment:
This is a bit roundabout. A better way to do this would be something like:
`static enum VALVE_STATE current_valve_state = VALVE_UNK;`
You can just update that every time you call an open/close function. This is a bit more descriptive and easier to reason about. It has the added benefit that it's consistent with the rest of the injector code.